cpp/tests/transform/nans_to_null_test.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/types.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <tests/utilities/base_fixture.hpp>
+#include <cudf/transform.hpp>
+#include <tests/utilities/type_lists.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+
+
+template <typename T>
+struct NaNsToNullTest : public cudf::test::BaseFixture {
+
+    void run_test(cudf::column input, cudf::column expected) {
+
+        auto got_mask = cudf::experimental::nans_to_nulls(input);
+        cudf::column got (static_cast<cudf::column>(input));
+        got.set_null_mask(std::move(*(got_mask.first)));
```

Comment:
I wonder if it would be a little cleaner for the return type of `nans_to_nulls` to be `std::pair<rmm::device_buffer, size_type>` (without the unique_ptr)... Maybe that makes it more difficult to avoid an accidental copy constructor though.

Reply:
Instead of using `set_null_mask` on a copy of the input column, you could maybe just construct a new column_view like so:
```
column_view(input.type(), input.size(), input.head(),
  got_mask.first.data().get(), got_mask.second);
```

Reply:
But as per the design requested in #3621 , `unique_ptr` was required. I think it is to be in sync with other APIs where we return with `unique_ptr`. @jrhemstad can please clarify.

Reply:
Yeah, Python usually likes us to return a heap allocated object as opposed to a stack allocated one. @shwina @kkraus14 do you care if the returned buffer is in a `unique_ptr` or if it's just returned as a stack allocated object?

Reply:
We're going to want to keep the buffer alive indefinitely via the smart pointer so heap allocated.